### PR TITLE
[wmi] extend tagging for related class & constants

### PR DESCRIPTION
**Pull request by yannmh**
Rebased #1253 on 5.2.0 agent release.

Added a few changes:
* String cast and flatten `link_value` tags values (https://github.com/DataDog/dd-agent/compare/yann/wmi-extend-tagging?expand=1#diff-7a46560288aed829df125f0734b355f9R89)
* Cast source property link value to `int` instead of `float` (experienced issues in my tests)

Thanks a lot @rlaveycal

### CODE REVIEW

File: conf.d/wmi_check.yaml.example [hunk 1]
```diff
@@ -35,6 +35,17 @@ instances:
   # WMI class you're using. This is only useful when you will have multiple
   # values for your WMI query. The examples below show how you can tag your
   # process metrics with the process name (giving a tag of "name:app_name").
+  #
+  # `constant_tags` optionally lets you tag each metric with a set of fixed values
+  #
+  # `link_tag` optionally specifies how to join to another property for tagging.
+  # Settings this will cause any instance number to be removed from tag_by values
```
Comment (LeoCavaille) [inline conf.d/wmi_check.yaml.example]: typo `Setting`

**Comment by LeoCavaille** [comment]
Thanks Yann, looking good!

Just two nitpicky comments, to make it look even nicer:
  * can you make the commit messages follow [these guidelines](https://github.com/DataDog/dd-agent/wiki/Conventions#github-titles)
    - short title are great to see in a git log (< 72 chars)
    - flags, like [wmi] are easy to grep for
  * can you keep the same spacing conventions in the YAML file it
    should be only 2 spaces comments

**Comment by yannmh** [comment]
Thanks @LeoCavaille !

**Comment by LeoCavaille** [comment]
Tried it on my box @yannmh , works well.
* One thing I am slightly worried about is the number of tags your example can create, on my simple box with nothing running on it: ~40 commandline and ~100 name tags.
Would this be possible to comment all examples in the sample file and let the user uncomment what he really wants?
Can we also remove the `allapps` example, it's not helpful?

* Second remark: looks like we cannot use several

**Comment by LeoCavaille** [comment]
Also, maybe reword the comment for `link_tag`
```
 # `link_tag` optionally specifies how to join to another property for tagging.
 # Setting this will cause any instance number to be removed from tag_by values
 # i.e. name:process#1 => name:process
 # `link_tag` is a list of four parameters: [param1, param2, param3, param4]
 #     param1: property of source that contains the link value
 #     param2: class to link to
 #     param3: property of target class to link to
 #     param4: property of target class that contains the value to tag with
```

**Comment by yannmh** [comment]
Thanks @LeoCavaille. I brought some changes following your feedback.

I am merging it when Travis tests are passed,

**Comment by LeoCavaille** [comment]
Léo Cavaillé tagged you in a RunKeeper activity
      
   .ExternalClass a span {
      text-decoration: none !important;
   }

   html
   {
      width: 100% !important;
      margin: 0 !important;
      padding: 0 !important;
      overflow-x: hidden !important; 
      -webkit-text-size-adjust: none !important;
      -ms-text-size-adjust: none !important;
   }

   body {
      width:100% !important;
      -webkit-text-size-adjust: none !important;
      -ms-text-size-adjust: none !important;
      margin:0 !important;
      padding:0 !important;
      font-family: 'Avenir', Helvetica, Arial, sans-serif;
      background-color: #25b7fa !important;
   }

   a span[class="bodylink"] {
      color: #1596d2 !important;
      text-decoration: underline !important;
   }

   a:active span {
      color: #badff0 !important;
   }

   img {
      display:block !important;
   }

   p {
      margin: 0 0 1.6em 0 !important;
      line-height: 150% !important;
   }

   h1 {
      line-height: 150% !important;
      font-size: 19pt !important;
      font-weight: bold !important;
   }

   table[class="w"] {
      margin:0 !important;
      padding:0 !important;
      width:100% !important;
      background-color: #25b7fa;
   }

   table[class="h"] {
      margin:0 auto !important;
      width:100% !important;
      height: 40px !important;
      font-size: 8pt !important;
   }

   table[class="c top"] p {
      line-height: 6pt !important;
      font-size: 8pt !important;
   }

   table[class="c top"] td span {
      font-size: 0 !important;
   }

   table[class="c top"] td a span {
      font-size: 8pt !important;
   }

   table[class="logo"] {
      margin:0 !important;
      padding:0 !important;
      width: 128px !important;
      height: 26px !important;
      padding-top: 2px !important;
      background-image: url('http://static-email.runkeeper.com/images/hlogo.png') !important;
      background-position: top left !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_facebook_icon_white"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/facebook_white.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_twitter_icon_white"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/twitter_white.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_instagram_icon_white"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/instagram_white.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_facebook_icon_blue"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/facebook_blue.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_twitter_icon_blue"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/twitter_blue.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   
   table[class="social_instagram_icon_blue"] {
      margin:0 !important;
      padding:0 !important;
      width: 40px !important;
      height: 40px !important;
      background-image: url('http://static-email.runkeeper.com/social_share_icons/40/instagram_blue.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }

   table[class="c"],
   table[class="c top"] {
      margin:0 !important;
      padding:0 !important; 
      width:600px !important;
      min-width:0 !important;
   }

   table[class="c top"] {
      height: 40px !important;
      font-size: 8pt !important;
   }

   a[class="button"],
   a[class="share_this_button facebook_button"],
   a[class="share_this_button twitter_button"] {
      text-decoration: none !important;
      width: 200px !important;
      height: 44px !important;
      padding: 0 !important;
      font-size: 14pt !important;
      font-weight: 500 !important;
      display: block !important;
      color: #FFFFFF !important;
      background-color: #1ccc66 !important;
      -webkit-box-shadow:  0px 3px 0px 0px rgba(25, 181, 90, 1) !important;
      box-shadow:  0px 3px 0px 0px rgba(25, 181, 90, 1) !important;
      -webkit-border-radius: 4px !important;
      border-radius: 4px !important;
      text-shadow: 0px 2px 0px #19b55a !important;
      filter: dropshadow(color=#19b55a, offx=0, offy=2) !important;
      line-height: 35pt !important;
      text-align: center !important;
   }

   a[class="share_this_button facebook_button"],
   a[class="share_this_button twitter_button"] {
      width: auto !important;
      height: auto !important;
      line-height: normal !important;
      padding: 10px 8px 10px 8px !important;
   }

   a[class="share_this_button facebook_button"] {
      background-color: #5074b4 !important;
      text-shadow: 0px 2px 0px #5074b4 !important;
      filter: dropshadow(color=#5074b4, offx=0, offy=2) !important;
      -webkit-box-shadow:  0px 3px 0px 0px rgba(63, 94, 148, 1) !important;
      box-shadow:  0px 3px 0px 0px rgba(63, 94, 148, 1) !important;
   }

   a[class="share_this_button twitter_button"] {
      background-color: #2ba9e1 !important;
      text-shadow: 0px 2px 0px #2ba9e1 !important;
      filter: dropshadow(color=#2ba9e1, offx=0, offy=2) !important;
      -webkit-box-shadow:  0px 3px 0px 0px rgba(33, 147, 198, 1) !important;
      box-shadow:  0px 3px 0px 0px rgba(33, 147, 198, 1) !important;
   }

   a[class="button"] span,
   a[class="share_this_button facebook_button"] span,
   a[class="share_this_button twitter_button"] span {
      color: #FFFFFF !important;
      text-decoration: none !important;
   }

   a[class="button"]:hover {
      background-color: #27dc73 !important;
   }

   a[class="share_this_button facebook_button"]:hover {
      background-color: #3f5e94 !important;
   }

   a[class="share_this_button twitter_button"]:hover {
      background-color: #2193c6 !important;
   }

   a[class="button"]:active {
      background-color: #1ccc66 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      position: relative !important;
      top: 3px !important;
      color: #93e2b4 !important;
   }

   a[class="share_this_button facebook_button"]:active {
      background-color: #3f5e94 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(63, 94, 148, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(63, 94, 148, 1) !important;
      position: relative !important;
      top: 3px !important;
   }

   a[class="share_this_button twitter_button"]:active {
      background-color: #2193c6 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(33, 147, 198, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(33, 147, 198, 1) !important;
      position: relative !important;
      top: 3px !important;
   }

   a[class="button"]:active span {
      color: #93e2b4 !important;
   }

   a[class="button facebook"] {
      text-decoration: none !important;
      width: 120px !important;
      height: 32px !important;
      padding: 0 !important;
      font-size: 11pt !important;
      font-weight: 500 !important;
      display: block !important;
      background-color: #5074b4 !important;
      -webkit-box-shadow:  0px 2px 0px 0px rgba(57, 90, 150, 1) !important;
      box-shadow:  0px 2px 0px 0px rgba(57, 90, 150, 1) !important;
      -webkit-border-radius: 4px !important;
      border-radius: 4px !important;
      text-shadow: 0px 2px 0px #395a96 !important;
      filter: dropshadow(color=#395a96, offx=0, offy=2) !important;
      text-align: center !important;
      line-height: 34px !important;
   }

   a[class="button facebook"] span {
      color: #FFFFFF !important;
      text-decoration: none !important;
   }

   a[class="button facebook"]:hover {
      background-color: #6288cb !important;
   }

   a[class="button facebook"]:active {
      background-color: #5074b4 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      position: relative !important;
      top: 3px !important;
      color: #d1e5f5 !important;
   }

   a[class="button facebook"]:active span {
      color: #d1e5f5 !important;
   }

   a[class="button twitter"] {
      text-decoration: none !important;
      width: 120px !important;
      height: 32px !important;
      padding: 0 !important;
      display: block !important;
      font-size: 11pt !important;
      font-weight: 500 !important;
      background-color: #2BA9E1 !important;
      -webkit-box-shadow:  0px 2px 0px 0px rgba(37, 128, 198, 1) !important;
      box-shadow:  0px 2px 0px 0px rgba(37, 128, 198, 1) !important;
      -webkit-border-radius: 4px !important;
      border-radius: 4px !important;
      text-shadow: 0 2px 0 #079bd1 !important;
      filter: dropshadow(color=#079bd1, offx=0, offy=2) !important;
      text-align: center !important;
      line-height: 34px !important;
   }

   a[class="button twitter"] span {
      color: #FFFFFF !important;
      text-decoration: none !important;
   }

   a[class="button twitter"]:hover {
      background-color: #4cb7e7 !important;
   }

   a[class="button twitter"]:active {
      background-color: #2BA9E1 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(37, 128, 198, 1) !important;
      position: relative !important;
      top: 3px !important;
      color: #d1e5f5 !important;
   }

   a[class="button twitter"]:active span {
      color: #d1e5f5 !important;
   }

   a[class="button primary"] span {
      color: #ffffff !important;
      text-decoration: none !important;
   }

   a[class="button primary"]:hover {
      background-color: #60db93 !important;
   }
   a[class="button primary"]:hover span {
      background-color: #60db93 !important;
   }

   a[class="button primary"]:active {
      background-color: #60db93 !important;
      -webkit-box-shadow:  0px 0px 0px 0px rgba(96,219,147, 1) !important;
      box-shadow:  0px 0px 0px 0px rgba(96,219,147, 1) !important;
      position: relative !important;
      top: 3px !important;
      color: #ffffff !important;
   }

   a[class="button primary"]:active span {
      color: #ffffff !important;
   }

   a[class="button"]:visited {
      color: #FFFFFF !important;
   }

   table[class="share"] {
      padding-top: 20px !important;
      padding-bottom: 20px !important;
   }

   table[class="footer"] {
      margin:0 !important;
      padding:0 !important;
      padding-top: 30px !important;
      width:100% !important;
      padding-bottom: 40px !important;
      font-size: 8pt !important;
      line-height: 70% !important;
   }

   table[class="footer"] p {
      font-size: 8pt !important;
      line-height: 70% !important;
      color: #e9f8ff !important;
   }

   table[class="upsell_img"] {
      margin:0 !important;
      padding:0 !important;
      width: 141px !important;
      height: 161px !important;
      background-size: 141px 161px !important;
      background-image: url('http://static-newsletter.runkeeper.com/elite_upsell/elite-clipboard.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   table[class="upsell_img"] td {
      width: 141px !important;
      height: 161px !important;
   }

      body {
         color: #ffffff !important;
      }
      table[class="c top"] {
         color: #e9f8ff !important;
      }


   /* iPhone, Android MDPI */
   @media only screen and (max-device-width:640px){
      p {
         font-size: 11pt !important;
         line-height: 120% !important;
      }

      h1 {
         line-height: 120% !important;
      }

      table[class="w"] {
         width: 100% !important;
      }

      table[class="c"],
      table[class="c top"] {
         width: 100% !important;
         margin-top: 0px !important;
      }
      table[class="cm"] {
         width: 92% !important;
         margin-top: 0px !important;
         margin: 0 auto !important;
      }
      
      table[class="social_share_icons"] {
         width: 130px !important;
         margin-top: 0px !important;
         margin: 0 auto !important;
      }

      a[class="button"] {
         width: 100% !important;
      }

      a[class="button primary"] {
         width: 200px !important;
      }

      table[class="footer"] {
         width: 100% !important;
      }

      table[class="footer"] p {
         width: 100% !important;
         font-size: 9pt !important;
         line-height: 150% !important;
      }

      table[class="footer"] p br {
         display: none !important;
      }

      table[class="share"] {
         width: 100% !important;
      }

      table[class="share"] td {
         min-width: 12px !important;
      }

      strong {
         font-weight:900 !important;
      }

   table[class="upsell_img"] {
      margin:0 !important;
      padding:0 !important;
      width: 141px !important;
      height: 161px !important;
      background-size: 141px 161px !important;
      background-image: url('http://static-newsletter.runkeeper.com/elite_upsell/elite-clipboard.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   table[class="upsell_img"] td {
      width: 141px !important;
      height: 161px !important;
   }

      
   }

   @media only screen and (max-device-width:320px) {
      table[class="logo"] {
         background-image: url('http://static-email.runkeeper.com/images/hlogo-ldpi.png') !important;
      }
      
      table[class="social_facebook_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/facebook_white.png') !important;
      }
      
      table[class="social_instagram_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/instagram_white.png') !important;
      }
      
      table[class="social_twitter_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/twitter_white.png') !important;
      }
      
      table[class="social_facebook_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/facebook_blue.png') !important;
      }
      
      table[class="social_instagram_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/instagram_blue.png') !important;
      }
      
      table[class="social_twitter_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/40/twitter_blue.png') !important;
      }
      
      td[class="ctd"] {
         padding:0 !important;
         padding-left: 0 !important;
         padding-right: 0 !important;
      }

      table[class="twocol-equalwidth-table"],
      table[class="twocol-table"] {
         width: 100% !important;
         margin-top: 0px !important;
         margin: 0 auto !important;
      }

   table[class="upsell_img"] {
      margin:0 !important;
      padding:0 !important;
      width: 75px !important;
      height: 86px !important;
      background-size: 75px 86px !important;
      background-image: url('http://static-newsletter.runkeeper.com/elite_upsell/elite-clipboard-sm.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   table[class="upsell_img"] td {
      width: 75px !important;
      height: 86px !important;
   }

      
   }

   @media (-webkit-min-device-pixel-ratio : 1.5) and (min-device-pixel-ratio : 1.5) {
      table[class="logo"] {
         background-image: url('http://static-email.runkeeper.com/images/hlogo-hdpi.png') !important;
         background-size: 128px 26px !important;
      }
      
      table[class="social_facebook_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/facebook_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_instagram_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/instagram_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_twitter_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/twitter_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_facebook_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/facebook_blue.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_instagram_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/instagram_blue.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_twitter_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/twitter_blue.png') !important;
         background-size: 40px 40px !important;
      }
      
      
   }

   @media (-webkit-min-device-pixel-ratio : 2) and (min-device-pixel-ratio : 2) {
      table[class="logo"] {
         background-image: url('http://static-email.runkeeper.com/images/hlogo@2x.png') !important;
         background-size: 128px 26px !important;
      }
      
     table[class="social_facebook_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/facebook_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_instagram_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/instagram_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_twitter_icon_white"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/twitter_white.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_facebook_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/facebook_blue.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_instagram_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/instagram_blue.png') !important;
         background-size: 40px 40px !important;
      }
      
      table[class="social_twitter_icon_blue"] {
         background-image: url('http://static-email.runkeeper.com/social_share_icons/80/twitter_blue.png') !important;
         background-size: 40px 40px !important;
      }

   table[class="upsell_img"] {
      margin:0 !important;
      padding:0 !important;
      width: 141px !important;
      height: 161px !important;
      background-size: 141px 161px !important;
      background-image: url('http://static-newsletter.runkeeper.com/elite_upsell/elite-clipboard@2x.png') !important;
      background-position: center center !important;
      background-repeat: no-repeat !important;
   }
   table[class="upsell_img"] td {
      width: 141px !important;
      height: 161px !important;
   }
      
      
   }
      
      
   
          
   
      
         
         
            
               
                  
                     
                        
                           
                              
                                    
                                       
                                       
                                       
                                          
                                              
                                          
                                       
                                       
                                       
                                    
                              
                              Léo Cavaillé tagged you in a RunKeeper activity
                           
                        
                     
                  
               
            
         
          
            

               
                  
                     
                        
                           
                              

          

      
         

      Léo Cavaillé tagged you in an activity to show that you worked out together.

         
      
          
            

      

              New to RunKeeper? RunKeeper is a mobile app and online fitness community that helps you track activities, view your training progress and receive motivation and encouragement from friends and other users.    Sign up today
 to get started!

      
              
            
      

   
   Sign up

      

              

      

              Already a user?    Log in
 now so we can associate this activity with your account.

      
          

      
      
        
            
          
   
      
         
            
                
            
         
      
   

        
         
        

      

            Activity:  Running
            Distance:  8.45 km
            Duration:  45:52
            Avg Pace:  5:26
            Avg Speed:  11.05
            Climb:  253

      
        
      
    
      

      

          
   View Activity


      
      


      
         Happy Running!
      
      
         
      




                              
                           
                        
                     
                  
               

            
         
          
            
            
            
            
            
            
         
   
   
   
   

      
      
      
      
      

         
   
   
      
         
            
                
            
         
      
   

         

         
          
         

         
   
   
      
         
            
                
            
         
      
   

         

         
          
         

         
   
   
      
         
            
                
            
         
      
   

         

      
      
      

   
   
   
   
            
            
            
            
            
            
            
            Sent by FitnessKeeper, Inc. 60 Canal Street Boston, MA 02114.
            Don't want to receive emails from us? Unsubscribe here.
            Click here to update your email address.
